### PR TITLE
JAMES-1819 Add ContentTransferEncoding when converting a message from…

### PR DESCRIPTION
**Pull request by aduprat**
… JMAP to Mime

### CODE REVIEW

**Comment by rouazana** [comment]
:+1:

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/MIMEMessageConverter.java [hunk 1]
```diff
@@ -104,6 +106,7 @@ public MIMEMessageConverter() {
             messageBuilder.setBody(createMultipart(creationMessageEntry.getValue(), messageAttachments));
         } else {
             messageBuilder.setBody(createTextBody(creationMessageEntry.getValue()));
+            messageBuilder.setContentTransferEncoding(QUOTED_PRINTABLE);
```
Comment (mbaechler) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/MIMEMessageConverter.java]: It's a builder, you probably can chain the call

Comment (mbaechler) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/MIMEMessageConverter.java]: who is responsible for encoding the actual content into quoted-printable ?

**Comment by aduprat** [comment]
test this please

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/MIMEMessageConverter.java (hunk 1 above)
Comment (aduprat) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/MIMEMessageConverter.java]: This is the job of the writer, here is the methods chain:
DefaultMessageWriter.writeMessage.writeEntity.encodeStream

Comment (mbaechler) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/MIMEMessageConverter.java]: let's add a test with a non-ascii character to ensure that

Comment (aduprat) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/MIMEMessageConverter.java]: Done

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/MIMEMessageConverterTest.java [hunk 2]
```diff
@@ -446,4 +446,27 @@ public void convertToMimeShouldAddAttachmentAndMultipartAlternativeWhenOneAttach
...
         assertThat(attachmentPart.getHeader().getField("Content-ID").getBody()).isEqualTo(expectedCID);
     }
+
+    @Test
+    public void convertShouldEncodeWhenNonASCIICharacters() {
+        // Given
+        MIMEMessageConverter sut = new MIMEMessageConverter();
+
+        CreationMessage testMessage = CreationMessage.builder()
+                .mailboxIds(ImmutableList.of("dead-bada55"))
```
Comment (mbaechler) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/MIMEMessageConverterTest.java]: please create a builder method that doesn't force you to create a singleton list

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/MIMEMessageConverterTest.java [hunk 3]
```diff
@@ -446,4 +446,27 @@ public void convertToMimeShouldAddAttachmentAndMultipartAlternativeWhenOneAttach
...
+                .subject("subject")
+                .from(DraftEmailer.builder().name("sender").build())
+                .htmlBody("Some non-ASCII characters: áÄÎßÿ")
+                .build();
+
+        // When
+        byte[] convert = sut.convert(new ValueWithId.CreationMessageEntry(
```
Comment (mbaechler) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/MIMEMessageConverterTest.java]: what is this empty list ?

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/MIMEMessageConverterTest.java [hunk 4]
```diff
@@ -446,4 +446,27 @@ public void convertToMimeShouldAddAttachmentAndMultipartAlternativeWhenOneAttach
...
+                CreationMessageId.of("user|mailbox|1"), testMessage), ImmutableList.of());
+
+        String expectedEncodedContent = "Some non-ASCII characters: =C3=A1=C3=84=C3=8E=C3=9F=C3=BF";
+
+        // Then
+        String actual = new String(convert, Charsets.UTF_8);
```
Comment (mbaechler) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/MIMEMessageConverterTest.java]: supposed to be ASCII, don't you think ?

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/MIMEMessageConverterTest.java [hunk 5]
```diff
@@ -446,4 +446,27 @@ public void convertToMimeShouldAddAttachmentAndMultipartAlternativeWhenOneAttach
...
+       assertThat(actual).contains(expectedEncodedContent);
```
Comment (mbaechler) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/MIMEMessageConverterTest.java]: indent bug

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/MIMEMessageConverterTest.java (hunk 2 above)
Comment (aduprat) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/MIMEMessageConverterTest.java]: OK

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/MIMEMessageConverterTest.java (hunk 3 above)
Comment (aduprat) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/MIMEMessageConverterTest.java]: OK

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/MIMEMessageConverterTest.java (hunk 4 above)
Comment (aduprat) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/MIMEMessageConverterTest.java]: Right

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/MIMEMessageConverterTest.java (hunk 5 above)
Comment (aduprat) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/MIMEMessageConverterTest.java]: OK

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/model/CreationMessage.java [hunk 6]
```diff
@@ -96,6 +96,11 @@ public Builder mailboxIds(List<String> mailboxIds) {
             return this;
         }
 
+        public Builder mailboxId(String mailboxId) {
```
Comment (mbaechler) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/model/CreationMessage.java]: hmm, I looks really useless, there's the ellipsis version already

Comment (aduprat) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/model/CreationMessage.java]: Right

**Comment by rouazana** [comment]
test this please

**Comment by mbaechler** [comment]
:+1:

**Comment by rouazana** [comment]
:+1:

**Pull request merged**